### PR TITLE
Add FrameDispatchCoordinator class and tests

### PR DESCRIPTION
**Pull request by harryz2000**

### CODE REVIEW

File: src/FrameDispatchCoordinator.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+					int aUpdateRefsPacketLateThresholdMs, 
+					std::chrono::milliseconds aUpdateRefsStepPacketEarlyMs) :
+	frameDelayCalculator(aUpdateRefsPacketEarlyThresholdMs, aUpdateRefsPacketLateThresholdMs, aUpdateRefsStepPacketEarlyMs)
+{
+}
+
+void FrameDispatchCoordinator::OnFrame(std::chrono::milliseconds now, uint64_t ts, uint64_t clockRate, MediaFrameListenerBridge& listenerBridge)
+{	
+	// Use the address of the MediaFrameListenerBridge as stream identifier
+	auto delayMs = frameDelayCalculator.OnFrame(uint64_t(&listenerBridge), now, ts, clockRate);
```
Comment (murillo128) [inline src/FrameDispatchCoordinator.cpp:13]: use `listenerBridge.GetMediaSSRC()` we will need to make sure that the ssrcs are unique when constructing them on the SRT server

File: test/unit/TestFrameDispatchCoordinator.cpp [hunk 2]
```diff
@@ -0,0 +1,2484 @@
+#include <gtest/gtest.h>
+
+#include "FrameDispatchCoordinator.h"
```
Comment (murillo128) [inline test/unit/TestFrameDispatchCoordinator.cpp:3]: can we change the test to be for the FrameDelayCalculator instead?

**Review by murillo128 — CHANGES_REQUESTED** [review]

File: src/FrameDispatchCoordinator.cpp (hunk 1 above)
Comment (harryz2000) [reply src/FrameDispatchCoordinator.cpp:13]: Sure.

File: test/unit/TestFrameDispatchCoordinator.cpp (hunk 2 above)
Comment (harryz2000) [reply test/unit/TestFrameDispatchCoordinator.cpp:3]: Yes. The current tests are actually for both FrameDispatchCoordinator (include FrameDelayCalculator)  and MediaFrameListenerBridge dispatching.  I will do a refactor to split some tests for FrameDelayCalculator and keep the tests for MediaFrameListenerBridge in the original test class.

Comment (harryz2000) [reply test/unit/TestFrameDispatchCoordinator.cpp:3]: Done.

File: src/FrameDispatchCoordinator.cpp (hunk 1 above)
Comment (harryz2000) [reply src/FrameDispatchCoordinator.cpp]: Done. The SrtConnection creates the audio and video listener bridge respectively with ssrc 1 and 2.

**Pull request merged**